magstock/__init__.py

```diff
@@ -0,0 +1,75 @@
+from uber.common import *
+from magstock._version import __version__
+
+config = parse_config(__file__)
+static_overrides(join(config['module_root'], 'static'))
+template_overrides(join(config['module_root'], 'templates'))
+
+
+@Config.mixin
+class ExtraConfig:
+    @property
+    def FOOD_COUNT(self):
+        with Session() as session:
+            return session.query(Attendee).filter_by(purchased_food=True).count()
+
+    @property
+    def PREREG_DONATION_OPTS(self):
+        """
+        We're overriding this so that we can cut off Supporter registrations
+        after we've sold a capped number and so that we can cut off shirt sales
+        after the supporter deadline (MAGStock doesn't assume inventory risk on
+        tshirts so they only sell them to people who preorder and do not sell
+        any on-site).
+        """
+        if self.AFTER_SUPPORTER_DEADLINE:
+            return [(amt, desc) for amt, desc in self.DONATION_TIER_OPTS if amt < self.SHIRT_LEVEL]
+        elif not self.SUPPORTER_AVAILABLE:
+            return [(amt, desc) for amt, desc in self.DONATION_TIER_OPTS if amt < self.SUPPORTER_LEVEL]
+        else:
+            return self.DONATION_TIER_OPTS
+
+
+@Session.model_mixin
+class Attendee:
+    allergies      = Column(UnicodeText, default='')
+    coming_with    = Column(UnicodeText, default='')
+    coming_as      = Column(Choice(c.COMING_AS_OPTS), nullable=True)
+    site_type      = Column(Choice(c.SITE_TYPE_OPTS), nullable=True)
+    noise_level    = Column(Choice(c.NOISE_LEVEL_OPTS), nullable=True)
+    camping_type   = Column(Choice(c.CAMPING_TYPE_OPTS), nullable=True)
+    purchased_food = Column(Boolean, default=False)
+
+    @cost_property
+    def food_cost(self):
+        return c.FOOD_PRICE if self.purchased_food else 0
+
+    @presave_adjustment
+    def roughing_it(self):
+        if self.site_type == c.PRIMITIVE and self.ribbon == c.NO_RIBBON:
+            self.ribbon = c.ROUGHING_IT
+
+    @property
+    def addons(self):
+        return ['Food all weekend'] if self.purchased_food else []
+
+
```

Comment:
This is how we add new validations.  In this case I happened to just implement all of these checks as a single validation, though theoretically we could have done it differently.

There are two things not shown here because we didn't need to do them for MAGStock, but both of them are possible:
* We can remove existing validations.  For example, an event might decide it doesn't care about age information at all, in any way.  It could remove all of the birthday and age group validations.
* We can remove or modify validations which are only performed while filling out one of the preregistration forms (as opposed to the validations which are always performed even on admin forms).